### ISSUE TITLE
How to decide the weight of ssim loss term?

### ISSUE THREAD

**Issue by MingYang-buaa** [nianticlabs/monodepth2#255]
Hey, thanks for your outstanding work!

In your work, the weight of ssim loss term has been set to 0.85. That's a empirical number or has got some physical meaning?
Did you test increasing or decreasing the weight?
In other word, how to decide the weight of ssim loss term when i test some other network arch such UNet++?

**Comment by mdfirman** [comment]
Hi, thanks for the interest.

It's a fairly empirical number – it's been used in many 'depth from mono' papers, and I think it's a pretty good choice. I don't think you would need to change it if you decided to use a different architecture or dataset. Other hyperparameters e.g. batch size and learning rate might have a larger effect on the results.

**Comment by mrharicot** [comment]
Hi,
The value of 0.85 originally came from this paper https://arxiv.org/abs/1511.08861 where they studied using SSIM, L1 and L2 for image reconstruction losses and empirically found out that it performed well with 0.85

**Comment by MingYang-buaa** [comment]
@mrharicot @mdfirman Thanks for your response! 
I'm trying to replace UNet with UNet++(https://arxiv.org/abs/1807.10165) and monitor the performance in every epoch. I find that the loss is decreasing but the abs_rel had no change(in a high value). So I guess this is the cause of the SSIM loss.

**Comment by mdfirman** [comment]
The L1 loss on reprojection doesn't correspond to abs_rel loss on depth values. So I don't know that adjusting the SSIM+l1 weighting is going to be the cause of the problem here. 

Out of interest: What values for the 7 metrics do you get after training on UNet++?

**Comment by MingYang-buaa** [comment]
Yeah，I trained for 5 epoch but it seems that cannot to converge so i stopped it.
the loss down to 0.9 but the metrics value: abs_rel:0.538    sqr_rel: 7.833   rmse:11.485   rmse_log:0.624  a1 a2 a3:0.292  0.542 0.724.
Any suggestion?

**Comment by mdfirman** [comment]
Hmm – yes these don't look very good at all. I suspect it's some problem with how unet++ has been used with the network architecture. Are you sure you have set the output activations correctly? And are extracting the multiscale losses in the correct locations?

Sorry I can't help more. I'd be interested to know how you get on though.

Thanks

**Comment by MingYang-buaa** [comment]
Hey, I'm sorry for the late reply. 

For the first question, yes, I used the sigmoid function to activate disp maps.
And for the second, I think it's not. As you know, the Unet++ used "deep supervision" to output multi-stage feature rather than multi-scale feature. That's may be the question.

Thanks for your help.
